File: Java/Commandes/src/stock/Commande.java
package stock;

import java.time.LocalDate;
import java.util.ArrayList;

public class Commande {
    private int id;
    private LocalDate date;
    private boolean type = false; // 1 = express | 0 = normale
    private boolean payee;
    private Client client;

    private static int staticID = 0;

    public Commande() {
    }

    Commande(LocalDate date, boolean type, boolean payee, Client client) {
        this.id = staticID + 1;
        this.date = date;
        this.type = type;
        this.payee = payee;
        this.client = client;
    }

    public int GetID() {
        return id;
    }

    public LocalDate GetDate() {
        return date;
    }

    public void SetDate(LocalDate date) {
        this.date = date;
    }

    public boolean GetType() {
        return type;
    }

    public void SetType(boolean type) {
        this.type = type;
    }

    public boolean GetPyee() {
        return payee;
    }

    public void SetPayee(boolean payee) {
        this.payee = payee;
    }

    public Client GetClient() {
        return client;
    }

    public void SetClient(Client client) {
        this.client = client;
    }

    public void SetPayee(Client client) {
        this.client = client;
    }

    public double calculTotalComande(ArrayList<LigneCommande> ligneCommandeList) {
        double totale = 0;
        for (LigneCommande ligneCommande : ligneCommandeList) {
            totale = totale + ligneCommande.calculTotalProduit();
        }
        return totale;
    }
}

File: S1/php/Pj_08/classes/Commande.php
<?php
class Commande
{
    private $id = 0;
    private $date;
    private $type = false;
    private $payee = false;
    private $idClient;

    static private $idInc = 0;

    function __construct($date, $type, $payee, $idClient)
    {
        $this->id = self::$idInc;
        $this->date = $date;
        $this->type = $type;
        $this->payee = $payee;
        $this->idCLient = $idClient;
        self::$idInc++;
    }

    function __destruct()
    {
        self::$idInc--;
    }

    function GetID()
    {
        return $this->id;
    }

    function GetDate()
    {
        return $this->date;
    }
    function SetDate($date)
    {
        $this->date = $date;
    }

    function GetType()
    {
        return $this->type;
    }
    function SetType($type)
    {
        $this->type = $type;
    }


    function GetIdCLient()
    {
        return $this->idCLient;
    }
    function SetIdCLient($idCLient)
    {
        $this->idCLient = $idCLient;
    }
}
